internal/dataplane/kong_client.go

```diff
@@ -119,6 +123,9 @@ type KongClient struct {
 
 	// eventRecorder is used to record warning events for resource failures.
 	eventRecorder record.EventRecorder
+
+	// SHAs is a slice is configuration hashes send in last batch send.
+	SHAs []string
```

Comment:
How do we map the target Kong gateway instances and SHAs, given that SHAs are stored in an array? By the order of adminURLs in CLI options/env vars?

Reply:
These are only checked for equality. PTAL at https://github.com/Kong/kubernetes-ingress-controller/blob/282d01f7f8ed45ac749079c3959480082cfbea3a/internal/dataplane/kong_client.go#L627-L630. We retrieve all SHAs and sort them and check whether that has changed.

What I gather from your proposal is that we could decrease the number of requests to Kong instances which already have the desired config i.e. keep a mapping of SHA per instance and if that hasn't changed don't render and send config to it.

This might be viewed as a desired optimization but OTOH this would happen pretty rarely in my eyes: most of the time all instances will receive the config upon each iteration (by default: each 3s). And that "optimization" would have little benefit in regular setting.

If you disagree please create an issue and we can discuss it there.